pkg/kcp/provider/gcp/vpcpeering/client/cloudComputeClient.go

```diff
@@ -0,0 +1,117 @@
+package client
+
+import (
+	compute "cloud.google.com/go/compute/apiv1"
+	pb "cloud.google.com/go/compute/apiv1/computepb"
+	"context"
+	"fmt"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/kyma-project/cloud-manager/pkg/kcp/provider/gcp/cloudclient"
+	"google.golang.org/api/option"
+)
+
+/*
+required GCP permissions
+=========================
+  ** Creates the VPC peering connection
+  compute.networks.addPeering => https://cloud.google.com/compute/docs/reference/rest/v1/networks/addPeering
+  ** Removes the VPC peering connection
+  compute.networks.removePeering => https://cloud.google.com/compute/docs/reference/rest/v1/networks/removePeering
+  ** Gets the network (VPCs) in order to retrieve the peerings
+  compute.networks.get => https://cloud.google.com/compute/docs/reference/rest/v1/networks/get
+*/
+
+func NewClientProvider() cloudclient.SkrClientProvider[Client] {
+	return func(ctx context.Context, saJsonKeyPath string) (Client, error) {
+		c, err := compute.NewNetworksRESTClient(ctx, option.WithCredentialsFile(saJsonKeyPath))
+		if err != nil {
+			return nil, err
+		}
+		//defer c.Close()
+		return &networkClient{cnc: c}, nil
+	}
+}
+
+type networkClient struct {
+	cnc *compute.NetworksClient
+}
+
+type Client interface {
+	CreateVpcPeeringConnection(ctx context.Context, name *string, remoteVpc *string, remoteProject *string, importCustomRoutes *bool, kymaProject *string, kymaVpc *string) (pb.NetworkPeering, error)
+}
+
+func (c *networkClient) CreateVpcPeeringConnection(ctx context.Context, name *string, remoteVpc *string, remoteProject *string, importCustomRoutes *bool, kymaProject *string, kymaVpc *string) (pb.NetworkPeering, error) {
+	networkPeering := pb.NetworkPeering{}
+
+	kymaNetwork := getFullNetworkUrl(*kymaProject, *kymaVpc)
+	remoteNetwork := getFullNetworkUrl(*remoteProject, *remoteVpc)
+
+	//peering from kyma to remote vpc
+	peeringRequestFromKyma := &pb.AddPeeringNetworkRequest{
+		Network: *kymaVpc,
+		Project: *kymaProject,
+		NetworksAddPeeringRequestResource: &pb.NetworksAddPeeringRequest{
+			NetworkPeering: &pb.NetworkPeering{
+				Name:                 name,
+				Network:              &remoteNetwork,
+				ImportCustomRoutes:   importCustomRoutes,
+				ExchangeSubnetRoutes: to.Ptr(true),
```

Comment:
Do not use Azure SDK in GCP implementations. If you need *bool you can use "k8s.io/utils/pointer"